bson-kotlinx/src/main/kotlin/org/bson/codecs/kotlinx/BsonEncoder.kt

```diff
@@ -140,25 +140,25 @@ internal class DefaultBsonEncoder(
     }
 
     override fun <T> encodeSerializableValue(serializer: SerializationStrategy<T>, value: T) {
-        deferredElementName?.let {
-            if (value != null || configuration.explicitNulls) {
-                encodeName(it)
-                super.encodeSerializableValue(serializer, value)
-            } else {
-                deferredElementName = null
-            }
-        }
-            ?: super.encodeSerializableValue(serializer, value)
+        deferredElementHandler.with(
+            {
+                if (value != null || configuration.explicitNulls) {
```

Comment:
Just a point of clarification:

It's unclear to me why `value` is ever `null` here.  If it is, why isn't `encodeNullableSerializableValue` called instead?